libgalois/include/katana/Properties.h

```diff
@@ -527,52 +527,53 @@ class VectorRef {
     const T* begin() const { return const_iterator(&data_[0]); }
     T* end() { return iterator(&data_[size_]); }
     const T* end() const { return const_iterator(&data_[size_]); }
+
   private:
     T* data_;
     size_t size_;
 };
 
 
 template <typename T, size_t N>
-class VectorPropertyView {
+class ArrayPropertyView {
   public:
-    using value_type = VectorRef<T>;
-    using reference = VectorRef<T>&;
-    using const_reference = const VectorRef<T>&;
+    using value_type = ArrayRef<T>;
+    using reference = ArrayRef<T>&;
+    using const_reference = const ArrayRef<T>&;
 
-    static Result<VectorPropertyView> Make(arrow::LargeListArray& array){
+    static Result<ArrayPropertyView> Make(arrow::LargeListArray& array){
       auto double_array_pointer = std::static_pointer_cast<arrow::DoubleArray>(array.values());
-      return VectorPropertyView(array, internal::GetMutableValuesWorkAround<T>(double_array_pointer->data(), 1, 0));
+      return ArrayPropertyView(array, internal::GetMutableValuesWorkAround<T>(double_array_pointer->data(), 1, 0));
```

Comment:
Why does this hard-code `double`? Can't `T` be any POD?

Reply:
Yes, `T` can be any POD. Do I need to use a switch statement for this line:

```C++
auto double_array_pointer = std::static_pointer_cast<arrow::DoubleArray>(array.values());
```

and replace `arrow::DoubleArray` with other arrow arrays such as `arrow::Int64Array` based on `T` ?

and the similar for the following line in `ArrayProperty`

```C++
auto inner = dynamic_cast<arrow::DoubleBuilder*>(outer->value_builder());
 ```